ando/tools/generator/nwb2bidsgenerator.py

```diff
@@ -0,0 +1,214 @@
+        columns=['Species', 'ParticipantID', 'Sex', 'Birthdate', 'Age', 'Genotype', 'Weight'])
+    dataset_desc_json = None
+
+    dataset_path = Path(dataset_path)
+    file_count = 0
+    sessions_count = 0
+    sub_ses_dict = defaultdict(list)
+    for nwb_file in dataset_path.glob('**/*.nwb'):
+        file_count += 1
+        channels_df = pd.DataFrame(
+            columns=['channel_id', 'Contact_id', 'type', 'units', 'sampling_frequency',
+                     'unit_conversion_multiplier'])
+        contacts_df = pd.DataFrame(
+            columns=[
+                'x',
+                'y',
+                'z',
+                'impedance',
+                'contact_id',
+                'probe_id',
+                'Location'])
+        probes_df = pd.DataFrame(columns=['probeID', 'type'])
+
+        with NWBHDF5IO(str(nwb_file), 'r') as io:
+            nwbfile = io.read()
+
+            # subject info:
+            if nwbfile.subject is not None:
+                sb = nwbfile.subject
+                if sb.subject_id is not None:
+                    subject_label = f'sub-{sb.subject_id}'
+                else:
+                    subject_label = f'sub-{sb.date_of_birth.strftime("%Y%m%dT%H%M")}'
+                if not participants_df['ParticipantID'].str.contains(
+                        subject_label).any():
+                    participants_df.loc[len(participants_df.index)] = \
+                        [sb.species, subject_label, sb.sex[0] if sb.sex is not None else None,
+                         sb.date_of_birth, sb.age, sb.genotype, sb.weight]
+            else:
+                subject_label = 'sub-none'
+                if not participants_df['ParticipantID'].str.contains(
+                        subject_label).any():
+                    participants_df.loc[len(participants_df.index)] = \
+                        [None, subject_label, None, None, None, None, None]
+
+            # dataset info:
+            if dataset_desc_json is None:
+                dataset_desc_json = dict(
+                    InstitutionName=nwbfile.institution, InstitutionalDepartmentName=nwbfile.lab,
+                    Name='Electrophysiology', BIDSVersion='1.0.X',
+                    Licence='CC BY 4.0',
+                    Authors=[
+                        list(nwbfile.experimenter) if nwbfile.experimenter is not None else None])
+            # sessions info:
+            subject_path = output_path/subject_label
+            bep_sessions_path = subject_path/f'{subject_label}_sessions.tsv'
+            if not bep_sessions_path.exists():
+                print(f'writing for subject: {subject_label}')
+                sessions_df = pd.DataFrame(
+                    columns=['session_id', '#_trials', 'comment'])
+            else:
+                sessions_df = pd.read_csv(bep_sessions_path, sep='\t')
+            if nwbfile.session_id is not None:
+                session_label = f'ses-{nwbfile.session_id}'
```

Comment:
same alphanumeric constrain here for the `nwbfile.session_id`

Reply:
fixed https://github.com/INT-NIT/AnDO/pull/73/commits/43436e3a0c3f9a329fd0461953debce226aeee24